analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/AvoidUnderPosting.cs

```diff
@@ -144,4 +146,11 @@ type switch
 
     private static bool HasValidateNeverAttribute(ISymbol symbol) =>
         symbol.HasAttribute(KnownType.Microsoft_AspNetCore_Mvc_ModelBinding_Validation_ValidateNeverAttribute);
+
+    private static bool HasDefaultValue(SyntaxNode node) => node switch
+    {
+        ParameterSyntax { Default: not null } => true,
+        PropertyDeclarationSyntax { Initializer: not null } property => property.Parent.ParameterList() is null,
```

Comment:
Why `property.Parent.ParameterList() is null`? How are property initializers not relevant if there is a primary constructor?